styx-scheduler-service/src/test/java/com/spotify/styx/SchedulerTest.java

```diff
@@ -110,12 +112,16 @@
   @Mock WorkflowExecutionGate gate;
   @Mock StateManager stateManager;
   @Mock Storage storage;
+  @Mock ShardedCounter shardedCounter;
+  @Mock CounterSnapshot counterSnapshot;
 
   @Before
   public void setUp() {
     workflows = new HashMap<>();
     when(gate.executionBlocker(any()))
         .thenReturn(WorkflowExecutionGate.NO_BLOCKER);
+    when(shardedCounter.getCounterSnapshot(anyString())).thenReturn(counterSnapshot);
+    when(counterSnapshot.pickShardWithSpareCapacity(anyLong())).thenReturn(0);
```

Comment:
Might have been appropriate to refactor this test too in accordance with `counterHasSpareCapacity`.